util/collections/machine_filters.go

```diff
@@ -204,3 +215,22 @@ func MatchesKubernetesVersion(kubernetesVersion string) Func {
 		return *machine.Spec.Version == kubernetesVersion
 	}
 }
+
+// GetFilteredMachinesForCluster returns a list of machines that can be filtered or not.
+// If no filter is supplied then all machines associated with the target cluster are returned.
+func GetFilteredMachinesForCluster(ctx context.Context, c client.Client, cluster *clusterv1.Cluster, filters ...Func) (Machines, error) {
```

Comment:
Do we want to use a client within this package?

Reply:
it was already being used in the same func in the util package https://github.com/kubernetes-sigs/cluster-api/pull/4207/files#diff-a61d5e39f45fb5ed2dc1ef71fa708c8db480e9a396adb8a5d9d17b8e6ba1e186L129, this moves it to a more specific sub package. Would there be a better place to put it if we don't want any client usage in util/?

Reply:
What about moving to a helper.go file under utils/collections (It is a little bit confusing having this func into machine_filters)

Reply:
done